File: src/main/java/org/usfirst/irs1318/gamesim/engine/event/Event.java
package org.usfirst.irs1318.gamesim.engine.event;

import org.usfirst.irs1318.gamesim.engine.activities.Activity;

import java.time.Instant;

// TODO
public class Event {
    private final Instant time;
    private final Activity activity;

    public Event(Instant time, Activity activity) {
        this.time = time;
        this.activity = activity;
    }

    public Instant getTime() {
        return time;
    }

    public Activity getActivity() {
        return activity;
    }
}

File: src/main/scala/org/usfirst/irs1318/gamesim/engine/event/Event.scala
package org.usfirst.irs1318.gamesim.engine.event

import org.usfirst.irs1318.gamesim.engine.action.Action
import org.usfirst.irs1318.gamesim.engine.time.Time
import org.usfirst.irs1318.gamesim.engine.time.Time._

case class Event(time: Time, action: Action)

object Event {
  implicit object EventOrdering extends Ordering[Event] {
    override def compare(x: Event, y: Event): Int = TimeOrdering.compare(x.time, y.time)
  }
}
